apps/web/lib/utils.ts

```diff
@@ -62,3 +62,12 @@ export const toJson = (obj: any): Object | null => {
     return null;
   }
 };
+
+export const deepEqual = (obj1: any, obj2: any): boolean => {
```

Comment:
When I just select another field after loading the page and don't change anything, `deepEqual` seems to be false. Maybe it makes sense to use a third party library for this functionality as it can be quiet complex to do right?
Something like: https://lodash.com/docs/4.17.15#isEqual ?
But please make sure to import only the required library: 
```
import { isEqual } from 'lodash';
```

Reply:
Absolutely, @mattinannt, you've pinpointed the issue correctly. The logic error in my code led to the alert being displayed even when we simply selected another field. The alert was not triggered due to proper comparison but rather because the `useEffect` hook was missing the dependency array `}, [localSurvey, survey]);,` causing it to not have the most recent `localSurvey` for comparison.

In essence, the alert was always being shown on reload due to this oversight.

Furthermore, I've decided to replace the `deepEqual` function with the `isEqual` function from `lodash`. Although the problem wasn't related to `deepEqual`, I believe it's better to rely on a well-established third-party library for more complex operations. 😊

Reply:
https://github.com/formbricks/formbricks/assets/69904519/fcdd332d-43d8-438d-9019-2e2645adc3b7